frontend/src/metabase/query_builder/components/notebook/NotebookStepPreview.jsx

```diff
@@ -22,100 +22,82 @@ import {
 
 const PREVIEW_ROWS_LIMIT = 10;
 
-class NotebookStepPreview extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      question: this.getPreviewQuestion(props.step),
-    };
-  }
+const getPreviewQuestion = step => {
+  const { getPreviewQuery, stageIndex } = step;
+  const query = getPreviewQuery();
+  const limit = Lib.currentLimit(query, stageIndex);
+  const hasSuitableLimit = limit !== null && limit <= PREVIEW_ROWS_LIMIT;
+  const queryWithLimit = hasSuitableLimit
+    ? query
+    : Lib.limit(query, stageIndex, PREVIEW_ROWS_LIMIT);
 
-  refresh = () => {
-    this.setState({
-      question: this.getPreviewQuestion(this.props.step),
-    });
-  };
+  return Question.create()
+    .setQuery(queryWithLimit)
+    .setDisplay("table")
+    .setSettings({ "table.pivot": false });
+};
 
-  getPreviewQuestion(step) {
-    const { getPreviewQuery, stageIndex } = step;
-    const query = getPreviewQuery();
-    const limit = Lib.currentLimit(query, stageIndex);
-    const hasSuitableLimit = limit !== null && limit <= PREVIEW_ROWS_LIMIT;
-    const queryWithLimit = hasSuitableLimit
-      ? query
-      : Lib.limit(query, stageIndex, PREVIEW_ROWS_LIMIT);
+const NotebookStepPreview = ({ step, onClose }) => {
+  const [question, setQuestion] = useState(getPreviewQuestion(step));
```

Comment:
please use `useState(() => getPreviewQuestion(step));` to avoid creating preview questions on each render